src/lib/viewers/doc/DocBaseViewer.js

```diff
@@ -1345,6 +1351,8 @@ class DocBaseViewer extends BaseViewer {
 
         const { pagesCount } = this.pdfViewer;
 
+        this.cacheThumbnailsToggledState(this.thumbnailsSidebar.isOpen);
```

Comment:
If the user toggles the sidebar open and closed quickly, does it behave as expected?

Reply:
Hard to reproduce because the position of the button changes as the sidebar is toggled. But even triggering via keyboard, it behaves as expected